### PR TITLE
feat: add .NET func-hosted notification bot template

### PR DESCRIPTION
**Pull request by swatDong**
- Keep the same folder names as VSC - `notification-function-base`, `notification-trigger-http`, `notification-trigger-timer`.
- GettingStarted.txt added
- .gitignore added
- Enable this template

E2E TEST: N/A

### CODE REVIEW

File: templates/bot/csharp/notification-function-base/ProjectName.csproj.tpl [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="6.0.0" />
+    <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="4.0.1" />
+    <PackageReference Include="Microsoft.TeamsFx" Version="0.5.0-rc">
+      <!-- Exclude TeamsFx wwwroot static files which are for frontend only. -->
+      <ExcludeAssets>contentFiles</ExcludeAssets>
+    </PackageReference>
+  </ItemGroup>
+
+  <ItemGroup>
+		<Content Include="Resources/*.json">
```
Comment (dooriya) [inline templates/bot/csharp/notification-function-base/ProjectName.csproj.tpl:35]: nit: format these sections.

File: templates/bot/csharp/notification-function-base/GettingStarted.txt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+
+Quick start
+----------
+
+Before you run or Start Debugging for the first time, follow these steps:
+
+1. This is a Bot project, so first use a Command Prompt to run this command: ngrok http 5130.
+
+2. Select the Project > Teams Toolkit > Configure Microsoft Teams app menu in Visual Studio. When prompted, sign in using your M365 account.
```
Comment (dooriya) [inline templates/bot/csharp/notification-function-base/GettingStarted.txt:18]: Update the command name to keep consistent with the latest UI.

Comment (swatDong) [reply templates/bot/csharp/notification-function-base/GettingStarted.txt:18]: Waiting for the existing template change... [pending]

File: templates/bot/csharp/notification-function-base/GettingStarted.txt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+3. Select the Debug > Start Debugging menu in Visual Studio or press F5. Visual Studio will launch the app inside Microsoft Teams using a browser.
+
+4. Once Microsoft Teams is loaded, select the Add button to install your app in Teams.
+
+5. Post HTTP request to http://localhost:5130/api/notification to trigger notification. You can use any HTTP tool: command line, PowerShell, etc.
+
```
Comment (dooriya) [inline templates/bot/csharp/notification-function-base/GettingStarted.txt:25]: and `Postman`?

**Comment by swatDong** [comment]
@dooriya @IvanJobs @Siglud @yukun-dong @hund030 - I've merged to latest dotnet template (GettingStarted, gitignore, project folder...). Please help review.

**Review by yukun-dong — APPROVED** [review]

**Review by dooriya — APPROVED** [review]

**Pull request merged**